service/routes.py

```diff
@@ -140,18 +144,85 @@ class WishlistResource(Resource):
   PUT /{wishlist_id} - Update a Wishlist with the id
   DELETE /{wishlist_id} -  Deletes a Wishlist with the id
   """
+  # ------------------------------------------------------------------
+  # RETRIEVE A Wishlist
+  # ------------------------------------------------------------------
+  @api.doc('get_wishlists')
+  @api.response(404, 'Wishlist not found')
+  @api.marshal_with(wishlist_vo)
+  def get(self, wishlist_id):
+    """
+    List all products in a wishlist based on a wishlist_id
+    """
+    app.logger.info("Request to list products in a wishlist")
+    wishlist = Wishlist.find_by_id(wishlist_id)
+    wishlist_products = Product.find_all_by_wishlist_id(wishlist_id)
+    return WishlistVo(wishlist, wishlist_products).serialize(), status.HTTP_200_OK
 
-  def get(self):
-    pass
+  #------------------------------------------------------------------
+  # UPDATE AN EXISTING PET
+  #------------------------------------------------------------------
+  @api.doc('rename_wishlists')
+  @api.response(404, 'Wishlist not found')
+  @api.response(400, 'The posted Wishlist data was not valid')
+  @api.expect(rename_wishlist_model)
+  @api.marshal_with(full_wishlist_model, code = 200)
+  def put(self, wishlist_id):
+    """
+    Renames a wishlist
+    This endpoint will rename an existing wishlist
+    """
+    app.logger.info("Request to rename a wishlist")
+    wishlist = Wishlist.find_by_id(wishlist_id)
+    if not wishlist:
+      abort(status.HTTP_404_NOT_FOUND, "Wishlist with id '{}' was not found.".format(wishlist_id))
 
-  def post(self):
-    pass
+    app.logger.info('Payload = %s', api.payload)
+    data = api.payload
+
+    query_res = Wishlist.find_all_by_user_id(wishlist.user_id)
+    res = []
+    for i in query_res:
+      res.append(i['name'])
+
+    num = 0
+    wishlist_fields = wishlist.serialize()
+
+    while 1:
+      if num == 0:
+        if data['name'] in res:
+          num += 1
+        else:
+          break
```

Comment:
It is said in the TDD test report that this line hasn't been checked. But I think it will be checked in the normal situation, which is in the test files. Does anyone know why?